frontend/src/components/Events/EventDetails.jsx

```
@@ -80,11 +107,26 @@ const EventDetails = () => {
           {/* Event Actions (RSVP Buttons) */}
+          {/* RSVP Buttons */}
           <div className="p-4 border-b border-gray-200 flex space-x-4">
-            <button className="px-4 py-2 bg-blue-600 text-white rounded-md font-semibold hover:bg-blue-700 transition">
+            <button
+              onClick={() => handleRSVP("going")}
+              className={`px-4 py-2 rounded-md font-semibold transition ${
+                rsvpStatus === "going"
+                  ? "bg-blue-700 text-white"
+                  : "bg-gray-200 text-gray-800 hover:bg-gray-300"
+              }`}
+            >
```

Biome (1.9.4)

[error] Provide an explicit type prop for the button element.

The default type of a button is submit, which causes the submission of a form when placed inside a `form` element. This is likely not the behaviour that you want inside a React application.
Allowed button types are: submit, button or reset

(lint/a11y/useButtonType)